### PR TITLE
DEPNotify Progress Bar for macOS Download

### PR DESCRIPTION
**Pull request by andredb90**
An idea to show progress during macOS Download.

Using arguments when calling the function in order to start the appropriate progress determination for startoinstall, installinstallmacos or fetch-full-installer.

### CODE REVIEW

File: erase-install.sh [hunk 1]
```diff
@@ -273,10 +273,11 @@ dep_notify() {
 }
 
 dep_notify_progress() {
-    # function for DEPNotify to show progress while the installer is being prepared
+    # function for DEPNotify to show progress while the installer is being downloaded or prepared
     last_progress_value=0
     current_progress_value=0
 
+    if [ $1 = startoinstall ]; then
```
Comment (grahampugh) [inline erase-install.sh:280]: Can you switch this to startosinstall? I know it doesn't really matter but it bugs me 😀

File: erase-install.sh [hunk 2]
```diff
@@ -1587,7 +1637,7 @@ if [[ $erase == "yes" ]]; then
         dn_status="${!dialog_reinstall_status}"
         dn_icon="$dialog_erase_icon"
         dep_notify
-        dep_notify_progress &
+        dep_notify_progress startoinstall >/dev/null 2>&1 &
```
Comment (grahampugh) [inline erase-install.sh:1640]: same as before, of course

File: erase-install.sh [hunk 3]
```diff
@@ -1611,7 +1661,7 @@ elif [[ $reinstall == "yes" ]]; then
         dn_status="${!dialog_reinstall_status}"
         dn_icon="$dialog_reinstall_icon"
         dep_notify
-        dep_notify_progress &
+        dep_notify_progress startoinstall >/dev/null 2>&1 &
```
Comment (grahampugh) [inline erase-install.sh:1664]: and the final time

**Comment by grahampugh** [comment]
This is great stuff! I'm almost wondering about bundling in DEPNotify in the package as well as python, but maybe it makes the package unnecessarily large?

File: erase-install.sh (hunk 1 above)
Comment (grahampugh) [reply erase-install.sh:280]: Ah don't worry, I'll do it.

**Comment by andredb90** [comment]
Right :) feel free to make any adjustments. (and have my apologies for the lousy naming mistake there). 

The script is super versatile as it is. I assume JAMF users are using jamf-helper and don't necessarily have to deploy the package. It could be an advantage for others though. 
A fully baked solution without having to worry about installing other tools along with it sounds enticing to me. - Maybe a good question for the community :)

**Comment by grahampugh** [comment]
Yes but the Jamf users are going to want the progress bar now, so they will start using the `--depnotify` option if they aren't already :-) (I'm one of those people...). 

The only thing I don't really like about DEPNotify is that I can't add a second button to the confirmation dialog, so the user must press Cmd + Alt + C to cancel. But I don't use the Confirm option anyway...

**Comment by andredb90** [comment]
Makes sense (after taking a look at the jamfHelper dialogs)

Submitting a feature request for a second button maybe? Or, thinking out loud here, you could use a single "Cancel" button instead and set a countdown before reinstall/erase starts automatically.

**Comment by andredb90** [comment]
Would it make sense to update the DEPNotify status to "Download complete!" once the download has completed successfully? (Instead of letting it sit at 100% and do nothing) 

Not sure how far up this TODO is on your list: 

dn_finished="Download complete!" # TODO localize this message

But once localized, we could use it for a proper status update.

**Comment by grahampugh** [comment]
Sounds good, if it's true that it's complete once it reaches 100%. But I think the script kills DEPNotify once the download is complete (and then the full screen "preparing" window opens). Are you not seeing that?

**Pull request merged**